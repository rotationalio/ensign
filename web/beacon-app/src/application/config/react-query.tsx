import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient();

export { QueryClientProvider, queryClient };
