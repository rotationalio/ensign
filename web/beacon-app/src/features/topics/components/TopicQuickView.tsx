import { Loader } from '@rotational/beacon-core';
import { Suspense, useEffect, useState } from 'react';

import { QuickView } from '@/components/common/QuickView';
import { SentryErrorBoundary } from '@/components/Error';

import useFetchTopicStats from '../hooks/useFetchTopicStats';
import { getDefaultTopicStats, getTopicStatsHeaders } from '../utils';

interface TopicQuickViewProps {
  topicID: string;
}
function TopicQuickView({ topicID }: TopicQuickViewProps) {
  const { topicStats, error } = useFetchTopicStats(topicID);
  const [topicData, setTopicData] = useState<any>(getDefaultTopicStats()); // by default we will show empty values

  // using useEffect will avoid infinite loop
  useEffect(() => {
    if (topicStats) {
      setTopicData(topicStats);
    }
  }, [topicStats]);

  useEffect(() => {
    if (error) {
      setTopicData(getDefaultTopicStats());
    }
  }, [error]);

  return (
    <Suspense fallback={<Loader />}>
      <SentryErrorBoundary fallback={<div>Something went wrong</div>}>
        <div>
          <QuickView data={topicData} headers={getTopicStatsHeaders()} className="my-4" />
        </div>
      </SentryErrorBoundary>
    </Suspense>
  );
}

export default TopicQuickView;
