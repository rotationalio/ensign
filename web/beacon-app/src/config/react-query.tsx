import { MutationCache, QueryCache, QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient({
  // define default options for all queries
  defaultOptions: {
    queries: {
      staleTime: 0,
      cacheTime: 5 * 60 * 1000,
      retry: 3,
      retryDelay: (attemptIndex) => Math.min(1000 * 2 ** attemptIndex, 30000),
      refetchInterval: false,
      refetchIntervalInBackground: false,
      refetchOnMount: true,
      refetchOnReconnect: true,
      refetchOnWindowFocus: true,
    },
  },
});

const queryCache = new QueryCache({
  onError: (error) => {
    console.log(error);
  },
  onSuccess: (data) => {
    console.log(data);
  },
});
const mutationCache = new MutationCache({
  onError: (error) => {
    console.log(error);
  },
  onSuccess: (data) => {
    console.log(data);
  },
});
export { mutationCache, queryCache, queryClient, QueryClientProvider };
