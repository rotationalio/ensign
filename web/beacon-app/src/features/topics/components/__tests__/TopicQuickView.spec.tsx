import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import TopicQuickView from '../TopicQuickView';

// mock SentryErrorBoundary
vi.mock('SentryErrorBoundary', () => ({
  __esModule: true,
  default: ({ children }) => children,
}));

// mock Suspense
vi.mock('Suspense', () => ({
  __esModule: true,
  default: ({ children }) => children,
}));
// mock useFetchTopicStats
const renderComponent = (props: { topicID: string }) => {
  const queryClient = new QueryClient();
  const wrapper = ({ children }: any) => (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
  return render(<TopicQuickView {...props} />, { wrapper });
};

// mock Trans tag from lingui
vi.mock('@lingui/macro', () => ({
  t: (str) => str,
  Trans: ({ children }) => children,
}));

// vi.mock("src/features/topics/hooks/useFetchTopicStats", async () => {
//   const actual = await vi.importActual("src/features/topics/hooks/useFetchTopicStats");
//   return {
//     ...actual,
//     // your mocked methods
//   },
// })

// mock useFetchTopicStats return value

describe('TopicQuickView', () => {
  it('should render the component with default value', () => {
    const { container } = renderComponent({ topicID: '1' });
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('quick-view-card-0')).toHaveTextContent('0');
    expect(screen.getByTestId('quick-view-card-1')).toHaveTextContent('0');
    expect(screen.getByTestId('quick-view-card-2')).toHaveTextContent('0');
    expect(screen.getByTestId('quick-view-card-3')).toHaveTextContent('0');
  });
});
