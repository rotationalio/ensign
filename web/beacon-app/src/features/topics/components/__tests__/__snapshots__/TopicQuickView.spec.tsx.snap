// Vitest Snapshot v1

exports[`TopicQuickView > should render the component with default value 1`] = `
<div>
  <div>
    <h1
      class="mt-4 text-lg font-semibold"
    >
      Quick View
    </h1>
    <div
      class="grid grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-4 my-4"
    >
      <div
        class="flex h-[100px] w-full flex-col rounded-2xl py-4 px-8"
        data-cy="quick-view-card-0"
        data-testid="quick-view-card-0"
        style="background-color: rgb(236, 246, 255);"
      >
        <h5
          class="pt-1 pb-3 font-semibold"
        >
          Online Publishers
        </h5>
        <p
          class="text-2xl font-semibold"
        >
          0
           
        </p>
      </div>
      <div
        class="flex h-[100px] w-full flex-col rounded-2xl py-4 px-8"
        data-cy="quick-view-card-1"
        data-testid="quick-view-card-1"
        style="background-color: rgb(229, 236, 246);"
      >
        <h5
          class="pt-1 pb-3 font-semibold"
        >
          Online Subscribers
        </h5>
        <p
          class="text-2xl font-semibold"
        >
          0
           
        </p>
      </div>
      <div
        class="flex h-[100px] w-full flex-col rounded-2xl py-4 px-8"
        data-cy="quick-view-card-2"
        data-testid="quick-view-card-2"
        style="background-color: rgb(236, 246, 255);"
      >
        <h5
          class="pt-1 pb-3 font-semibold"
        >
          Total Events
        </h5>
        <p
          class="text-2xl font-semibold"
        >
          0
           
        </p>
      </div>
      <div
        class="flex h-[100px] w-full flex-col rounded-2xl py-4 px-8"
        data-cy="quick-view-card-3"
        data-testid="quick-view-card-3"
        style="background-color: rgb(229, 236, 246);"
      >
        <h5
          class="pt-1 pb-3 font-semibold"
        >
          Data Storage
        </h5>
        <p
          class="text-2xl font-semibold"
        >
          0
           
          GB
        </p>
      </div>
    </div>
  </div>
</div>
`;
